### ISSUE TITLE
Feature request: zoom to bounds of cluster on click

### ISSUE THREAD

**Issue by rhewitt22** [alex3165/react-mapbox-gl#333]
When the user clicks on a cluster the map zooms to the bounds of the cluster.

Similar to how [leaflet.markercluster](https://github.com/Leaflet/Leaflet.markercluster) works.

[Example](https://leaflet.github.io/Leaflet.markercluster/example/marker-clustering-realworld.388.html).

**Comment by alex3165** [comment]
Hey @rhewitt22 , I think that could be possible, how would you see it in term of API? 

What about something like:
```
<Cluster ClusterMarkerFactory={...} zoomOnClick={true}>
...
</Cluster>
```

cc @Wykks

**Comment by rhewitt22** [comment]
@alex3165 it doesn't get any simpler of an API than that; I like it.
😃

**Comment by alex3165** [comment]
Merged and released with version `2.5.0`
